Sources/SwiftUINavigation/Inspector.swift

```diff
@@ -72,4 +68,25 @@
       }
     }
   }
+
+  @available(iOS 17, macOS 14, *)
+  @available(tvOS, unavailable)
+  @available(watchOS, unavailable)
+  @available(visionOS, unavailable)
+  private struct InspectorModifier<Item, Destination: View>: ViewModifier {
+    @Binding var item: Item?
+    let destination: (Binding<Item>) -> Destination
+    @State var cachedItem: Item?
+
+    func body(content: Content) -> some View {
+      content.inspector(isPresented: Binding($item)) {
+        if let defaultValue = item ?? cachedItem {
+          destination(Binding(unwrapping: $item, default: defaultValue))
+            .onAppear {
```

Comment:
I actually think we need something more like this to ensure the cached value stays up-to-date...

```suggestion
            .onChange(of: UUID()) {
```

@mbrandonw What do you think?